packages/enketo-core/src/widget/date/datepicker-extended.js

```diff
@@ -87,7 +87,12 @@ class DatepickerExtended extends Widget {
     _setChangeHandler($fakeDateI) {
         const { settings } = this;
 
-        $fakeDateI.on('change paste', (e) => {
+        let changeEvent = 'change';
+        if (!$fakeDateI.closest('label').hasClass('readonly')) {
+            changeEvent += ' paste';
+        }
+
+        $fakeDateI.on(changeEvent, (e) => {
```

Comment:
Thanks for noticing this and taking on a fix! This works but feels a bit hard to understand to me. 

No `change` event should ever be fired on a readonly date, right? Could we instead not register the handler at all in that case? Something like 

```
if (!$fakeDateI.closest('label').hasClass('readonly')) {
  $fakeDateI.on('change paste', (e) => {
    ...
}
```

Reply:
Got it. Adressed in the next commit.

Thanks.